modules/ncspot/hm.nix

```
@@ -1,34 +1,54 @@
-{ config, lib, ... }:
+{ mkTarget, lib, ... }:
+mkTarget {
+  name = "ncspot";
+  humanName = "Ncspot";
+  extraOptions = {
+    background = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      internal = true;
+      default = null;
+    };
+  };
 
-{
-  options.stylix.targets.ncspot.enable =
-    config.lib.stylix.mkEnableTarget "Ncspot" true;
-
-  config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.ncspot.enable)
+  configElements = [
+    (
+      { cfg }:
+      lib.mkIf (cfg.background != null) {
+        programs.ncspot.settings.theme = {
+          inherit (cfg) background;
+          playing_bg = cfg.background;
+        };
+      }
+    )
+    (
+      { opacity }:
+      lib.mkIf (opacity.terminal != 1.0) {
+        stylix.targets.ncspot.background = "#00000000";
+      }
```

Does configElements support wrapping the entire config in `mkIf`?

It should work if mkTarget is doing something like `config = mkMerge (map … configElements)`.

Either way, I wonder if pushing down the mkIf is more readable?


```suggestion
      {
        stylix.targets.ncspot.background = lib.mkIf (opacity.terminal != 1.0) "#00000000";
      }
```

?

---

> Either way, I wonder if pushing down the mkIf is more readable?

I think so.